torch/optim/_device_utils.py

```
@@ -0,0 +1,79 @@
+    """
+    # BUG E: Hardcoded device list doesn't include 'mps' which was added to
+    # _get_capturable_supported_devices() in optimizer.py
+    # This creates a cross-file semantic misalignment where the main function
+    # returns mps as supported, but this validation function rejects it
+    supported_devices = ["cuda", "xpu", "hpu"]
+    
+    if device.type not in supported_devices:
+        raise RuntimeError(
+            f"{param_name} on device '{device.type}' does not support capturable "
+            f"optimizer operations. Supported devices: {supported_devices}"
+        )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)